### ISSUE TITLE
New submissions of vasp jobs will kill the ones which submitted before

### ISSUE THREAD

**Issue by yuyuanjingxuan** [pyiron/pyiron_atomistics#407]
**Summary**
Hi all,

I recently found when a new vasp job stated running,  jobs which submitted previous were killed for unknown reasons. In my case, 20 cores were applied in each of jobs. I found only 6 of them could exist simultaneously in the queue. Attached are the error messages of jobs which was killed:
```bash
srun: error: cmti021: tasks 6,16: Killed
srun: Terminating job step 2517847.0
slurmstepd: error: *** STEP 2517847.0 ON cmti021 CANCELLED AT 2021-11-02T12:25:15 ***
srun: error: cmti021: task 10: Killed
forrtl: error (78): process killed (SIGTERM)
Image              PC                Routine            Line        Source             
vasp544_std_poten  0000000005ED608B  Unknown               Unknown  Unknown
libpthread-2.26.s  00002AFDD43C92D0  Unknown               Unknown  Unknown
libmpi.so.12.0.0   00002AFDD2ECAB1C  Unknown               Unknown  Unknown
libmpi.so.12.0.0   00002AFDD348347D  Unknown               Unknown  Unknown
libmpi.so.12.0.0   00002AFDD305566A  Unknown               Unknown  Unknown
libmpi.so.12.0.0   00002AFDD2DC0047  Unknown               Unknown  Unknown
libmpi.so.12.0.0   00002AFDD2E5002D  Unknown               Unknown  Unknown
libmpi.so.12.0.0   00002AFDD2E36D8E  Unknown               Unknown  Unknown
libmpi.so.12.0.0   00002AFDD2F53735  Unknown               Unknown  Unknown
libmpi.so.12.0.0   00002AFDD2DC0EBA  MPI_Alltoall          Unknown  Unknown
libmpifort.so.12.  00002AFDD29F513E  mpi_alltoall__        Unknown  Unknown
vasp544_std_poten  0000000000412B37  Unknown               Unknown  Unknown
vasp544_std_poten  000000000057F78A  Unknown               Unknown  Unknown
vasp544_std_poten  0000000000EC3560  Unknown               Unknown  Unknown
vasp544_std_poten  0000000000ED1324  Unknown               Unknown  Unknown
vasp544_std_poten  00000000014F79BA  Unknown               Unknown  Unknown
vasp544_std_poten  00000000014CBAED  Unknown               Unknown  Unknown
vasp544_std_poten  0000000000407512  Unknown               Unknown  Unknown
libc-2.26.so       00002AFDD493134A  __libc_start_main     Unknown  Unknown
vasp544_std_poten  000000000040742A  Unknown               Unknown  Unknown
```
I also tried to submitted jobs in different queues: `cm`, `cmti` and `cmfe`, while the issue still occurs. Additionally, this issue only happened on vasp jobs. I've tested lammps jobs and there is no error. It should not be the issues of input files and vasp executable. Do you suffer the similar issues as me?

**pyiron Version and Platform**
Pyrion 0.4.5
Python 3.8.12

**Expected Behavior**
No error messages.

**Actual Behavior**
The queue state shows
```bash
Nodes required for job are DOWN, DRAINED or reserved for jobs in higher priority partitions
```
Then, vasp jobs which submitted previous were killed when a new vasp job was running. Only 6 of them were allowed to run simultaneously in my case.

**Comment by pmrv** [comment]
We have some issues with random VASP crashes right now.  Have you verified that it is really the "seventh" vasp job that causes the crash or do the jobs also crash, when only six jobs are queued?

In my cases the crashes appear to be caused by high k-point sampling (~25 for small structures, less for larger structures). @prince-mathews and @usaikia might have other tips.

**Comment by prince-mathews** [comment]
So the problem is that when a new job starts running, one of the already running jobs gets killed, so only 6 run at a time. Am I right? 
Also, did any of the jobs finish?

**Comment by yuyuanjingxuan** [comment]
> We have some issues with random VASP crashes right now. Have you verified that it is really the "seventh" vasp job that causes the crash or do the jobs also crash, when only six jobs are queued?
> 
> In my cases the crashes appear to be caused by high k-point sampling (~25 for small structures, less for larger structures). @prince-mathews and @usaikia might have other tips.

Yes, I doublechecked it. The six jobs could run without any issues. Once the 7th one is running, the 1st job is killed.

**Comment by yuyuanjingxuan** [comment]
> So the problem is that when a new job starts running, one of the already running jobs gets killed, so only 6 run at a time. Am I right? Also, did any of the jobs finish?

Yes, it is. The 6 jobs are still running, and the output files are keeping updating. None of them are finished yet.

**Comment by usaikia** [comment]
In my case it was not the 6th job. More than six jobs were running at the same time without any problem. Some of them got killed with the same error as @yuyuanjingxuan mentioned after running successfully for a few ionic steps.

**Comment by samwaseda** [comment]
By the way it's not directly related to this issue, but it might be a good opportunity to let @freyso know what's missing in SPHInX and you can maybe start thinking about switching to it.

**Comment by pmrv** [comment]
> > We have some issues with random VASP crashes right now. Have you verified that it is really the "seventh" vasp job that causes the crash or do the jobs also crash, when only six jobs are queued?
> > In my cases the crashes appear to be caused by high k-point sampling (~25 for small structures, less for larger structures). @prince-mathews and @usaikia might have other tips.
> 
> Yes, I doublechecked it. The six jobs could run without any issues. Once the 7th one is running, the 1st job is killed.

This sounds like it's something about the queuing system and not pyiron as such.  Is it related to this specific six calculations or does the same thing happen with 6 cheap calculations?

**Comment by yuyuanjingxuan** [comment]
> In my case it was not the 6th job. More than six jobs were running at the same time without any problem. Some of them got killed with the same error as @yuyuanjingxuan mentioned after running successfully for a few ionic steps.

I guess maybe there was a limitation of maximum cores for one user. In my case, it is 120 cores (6 jobs).

**Comment by niklassiemer** [comment]
However, such limitations should only pause calculations to start running not killing already running ones in favor of starting a new one...

**Comment by yuyuanjingxuan** [comment]
Similar issue occurs on my LAMMPS jobs now. I found all my LAMMPS jobs which are running crashed simultaneously. One of them survived. Attached is the error messages:
```bash
srun: error: cmti005: task 1: Bus error
srun: Terminating job step 2519019.0 
slurmstepd: error: *** STEP 2519019.0 ON cmti005 CANCELLED AT 2021-11-04T12:34:18 ***
slurmstepd: error: *** STEP 2519019.0 ON cmti005 CANCELLED AT 2021-11-04T12:34:18 ***
srun: error: cmti005: tasks 0,2-19: Terminated 
srun: Force Terminated job step 2519019.0 
```
It seems that the issue caused by insufficient default memory. When the jobs ask for more, processors cannot attempt the memory access requested.
I guess the issue of VASP also caused by the memory problem. I performed the same jobs before with only gamma point. Everything was normal. However, all jobs crashed with denser kpoint meshes. It did not take place once the job was running, but occurred during running process.

**Comment by pmrv** [comment]
I'm running a lot of lammps jobs right now and can say it's at least not a general problem.  It really sounds like a [memory problem](https://ask.cyberinfrastructure.org/t/what-does-it-mean-when-i-get-a-bus-error-in-my-job/1101).  You can ask for more memory on your jobs with `job.server.memory_limit`.

**Comment by yuyuanjingxuan** [comment]
Yes, I've tried to ask for more memory, while it did not work. Then I increase the number of cores from 40 to 60 and 80. Jobs running with 60 core still exported the similar issue. Everything looks fine for those with 80 cores.

**Comment by niklassiemer** [comment]
I just do not get how these things are related... Each single simulation should be independent of all the others - latest after given to slurm to handle them. Since afterwards, each calculation should be running on their own nodes and with their own memory limitations. Therefore, it should not matter if I submit 3 or 30 jobs of the same kind for their run behavior... (Of course, slurm will distribute them according to free resources and configurations on how many nodes you are allowed to run on). Do I have some logic mistake here? Do we have other shared resources (except the database to report the status to) which might affect parallel running calculations?
